internal/juju/models.go

```diff
@@ -130,35 +130,47 @@ func (c *modelsClient) Create(name string, controller string, cloudList []interf
 	return &modelInfo, nil
 }
 
-func (c *modelsClient) Read(uuid string) (*string, *params.ModelInfo, error) {
-	conn, err := c.GetConnection(nil)
+func (c *modelsClient) Read(uuid string) (*string, *params.ModelInfo, map[string]interface{}, error) {
+	modelmanagerConn, err := c.GetConnection(nil)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
-	client := modelmanager.NewClient(conn)
-	defer client.Close()
+	modelconfigConn, err := c.GetConnection(&uuid)
+	if err != nil {
+		return nil, nil, nil, err
+	}
 
-	models, err := client.ModelInfo([]names.ModelTag{names.NewModelTag(uuid)})
+	modelmanagerClient := modelmanager.NewClient(modelmanagerConn)
+	defer modelmanagerClient.Close()
+
+	modelconfigClient := modelconfig.NewClient(modelconfigConn)
+	defer modelconfigClient.Close()
+
+	models, err := modelmanagerClient.ModelInfo([]names.ModelTag{names.NewModelTag(uuid)})
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	if len(models) > 1 {
-		return nil, nil, errors.New(fmt.Sprintf("more than one model returned for UUID: %s", uuid))
+		return nil, nil, nil, errors.New(fmt.Sprintf("more than one model returned for UUID: %s", uuid))
```

Comment:
Is this possible, can different models have the same uuid? 😟

Reply:
Probably not, it is supposed to be a UUID after all... but I figured it was safer not to assume. The alternative would be to always take element 0, but if there is some how a duplicate, we then end up targeting the wrong one 🤷‍♂️